### PR TITLE
updated login, projects and ci-cd pipeline

### PR DESCRIPTION
**Pull request by Abeinevincent**
Updated ci-cd workflow to enable it run on all operating systems supported by github hosted runners' virtual machines that is ubuntu-latest, windows-latest and macos-latest to ensure that code is reliable and compatible with a wide range of operating systems and to help catch any platform-specific issues or differences in behavior that may exist unknowingly.
Updated projects page by adding some padding to have more readable content.
Updated login component by adding error messages when no input is received from the user and marked all login input fields compulsory with an asterisk that is must all be entered and with correct values according to their type attribute.

### CODE REVIEW

File: components/Login.jsx [hunk 1]
```diff
@@ -49,21 +49,23 @@ const LoginSection = () => {
...
+            type='email'
             value={email}
+            required
             onChange={(e) => setEmail(e.target.value)}
             className='bg-transparent py-2 pl-8 font-medium border-b-[1px] border-b-slate-900 w-full outline-none'
           />
         </div>
         <div className='mt-6 relative'>
-          <h3>Password</h3>
+          <h3>Password<span style={{color:'red'}}>*</span></h3>
```
Comment (HassanBahati) [inline components/Login.jsx:63]: @Abeinevincent i think we should set a design system via [tailwind.config.js](https://github.com/OSCA-Kampala-Chapter/oscakampala.github.io/blob/main/tailwind.config.js) than adding custom colors directly in there.

File: components/Login.jsx [hunk 2]
```diff
@@ -49,21 +49,23 @@ const LoginSection = () => {
       <h1>Login here</h1>
       <form onSubmit={handleSubmit} className=''>
         <div className='mt-6 relative'>
-          <h3>Email or username</h3>
+          <h3>Email or username<span style={{color:'red'}}>*</span></h3>
```
Comment (HassanBahati) [inline components/Login.jsx:52]: @Abeinevincent i think we should set a design system via [tailwind.config.js](https://github.com/OSCA-Kampala-Chapter/oscakampala.github.io/blob/main/tailwind.config.js) than adding custom colors directly in there.

File: pages/projects.jsx [hunk 3]
```diff
@@ -3,7 +3,7 @@
 function projects() {
   return (
     <section className="bottom-50">
-      <div>
+      <div style={{ padding: '10px' }}>
```
[the remark this answers is not in the document]
Comment (Abeinevincent) [reply pages/projects.jsx:6]: Okay, Its truly fine @HassanBahati  .

File: components/Login.jsx (hunk 2 above)
Comment (Abeinevincent) [reply components/Login.jsx:52]: Thats okay surely @HassanBahati . However, we should atleast have an alternative to tailwind as not all devs willing to contribute to the design are well off with tailwind. I would thus suggest we also allow custom css using usual classname and a general css file for that to keep the codebase clean.

Comment (HassanBahati) [reply components/Login.jsx:52]: true @Abeinevincent. we should strive to cater for non-tailwind devs as they should be able to contribute to the project. Actually even no-code devs should be able to pick up quickly.
My comment here is towards inline css not custom css. Maintainability and cleanliness get handcuffed in such a scenario. We can have a css file from which classes can get joined to wherever.

Comment (Abeinevincent) [reply components/Login.jsx:52]: I get you surely @HassanBahati . A separate css file is a nice workout for a clean and more readable code.

**Review by HassanBahati — COMMENTED** [review]
Looks Good To me

**Comment by Abeinevincent** [comment]
Glad to hear @HassanBahati . Another quick one I am suggesting a refactor for the ci/cd pipeline code(yml file) to capture the current connections. It looked to me like the existing one had been connected to the previous host(github pages).

**Pull request merged**